engine/src/main/java/org/hibernate/search/engine/search/reference/FieldReference.java

```diff
@@ -0,0 +1,38 @@
+package org.hibernate.search.engine.search.reference;
+
+import org.hibernate.search.engine.search.common.ValueConvert;
+import org.hibernate.search.util.common.annotation.Incubating;
+
+/**
+ * @param <T> The expected returned type.
+ */
+@Incubating
+public interface FieldReference<T> {
+
+	String absolutePath();
+
+	default ValueConvert valueConvert() {
+		return ValueConvert.YES;
+	}
+
+	Class<T> type();
+
+	interface FieldAttributeReference<T, V> extends FieldReference<T> {
+
+		/**
+		 * Applies {@link org.hibernate.search.engine.search.common.ValueConvert#NO}
+		 */
+		FieldReference<V> noConverter();
+
+		/**
+		 * Applies {@link org.hibernate.search.engine.search.common.ValueConvert#PARSE}
+		 */
+
+		FieldReference<String> asString();
+	}
+
+	interface FieldObjectReference<T> extends FieldReference<T> {
+
+	}
```

Comment:
I definitely wouldn't make these nested interfaces FWIW; it makes code unnecessarily verbose.